src/main/java/com/juniorstart/juniorstart/service/ProjectService.java

```diff
@@ -0,0 +1,290 @@
+        ValidatePageUtil.validatePageNumberAndSize(page, size);
+
+        Pageable pageable = PageRequest.of(page, size, Sort.Direction.DESC, "createdAt");
+
+        Technologies technology = technologiesRepository.findById(technologyId)
+                .orElseThrow(() -> new ResourceNotFoundException("Technology", "technologyTitle", technologyId));
+        List<Long> longsOfProjects = technology.getProjects().stream().map(Project::getId).collect(Collectors.toList());
+        Page<Project> projects = projectRepository.findByIdIn(longsOfProjects, pageable);
+
+        if(projects.getNumberOfElements() == 0) {
+            return new PagedResponse<>(Collections.emptyList(), projects.getNumber(), projects.getSize(), projects.getTotalElements(), projects.getTotalPages(), projects.isLast());
+        }
+
+        List<ProjectResponse> projectResponses = projects.map(project -> {
+
+            User user = userDao.findByPublicId(project.getCreatedBy())
+                    .orElseThrow(() -> new ResourceNotFoundException("User", "userId", project.getCreatedBy()));
+
+            return ModelMapper.mapProjectToProjectResponse(project, user);
+
+        }).getContent();
+
+        return new PagedResponse<>(projectResponses, projects.getNumber(), projects.getSize(), projects.getTotalElements(), projects.getTotalPages(), projects.isLast());
+
+
+    }
+
+    /**
+     * Create a {@link Project} from {@link ProjectRequest}
+     * @param projectRequest A request with a credentials.
+     * @return A model ready to save.
+     */
+    private Project createProjectModel(ProjectRequest projectRequest) {
+        Project newProject = new Project();
+        newProject.setRepository(projectRequest.getRepository());
+        newProject.setName(projectRequest.getName());
+        newProject.setTitle(projectRequest.getTitle());
+        newProject.setNumberOfSeats(projectRequest.getNumberOfSeats());
+        newProject.setDescription(projectRequest.getDescription());
+        newProject.setBody(projectRequest.getBody());
+
+        return newProject;
+    }
+
+    /**
+     * Update a {@link Project} with {@link ProjectRequest}
+     * @param currentProject A current project's object.
+     * @param projectRequest A projectRequest that will update a project
+     * @return A updated project ready to save.
+     */
+    private Project updateProjectModel(Project currentProject, ProjectRequest projectRequest) {
+        currentProject.setRecruiting(projectRequest.isRecruiting());
+        currentProject.setBody(projectRequest.getBody());
+        currentProject.setDescription(projectRequest.getDescription());
+        currentProject.setNumberOfSeats(projectRequest.getNumberOfSeats());
+        currentProject.setTitle(projectRequest.getTitle());
+        currentProject.setName(projectRequest.getName());
+        currentProject.setRepository(projectRequest.getRepository());
+
```

Comment:
Builder

Reply:
Tu moim zdaniem nie opłaca się używać buildera, ponieważ edytujemy tylko kilka wartości zamiast tworzenia nowego obiektu itp.

Reply:
``currentProject.setRecruiting(projectRequest.isRecruiting());
        currentProject.setBody(projectRequest.getBody());
        currentProject.setDescription(projectRequest.getDescription());
        currentProject.setNumberOfSeats(projectRequest.getNumberOfSeats());
        currentProject.setTitle(projectRequest.getTitle());
        currentProject.setName(projectRequest.getName());
        currentProject.setRepository(projectRequest.getRepository());

 return currentProject;``

Reply:
Zobacz jak to o wiele czytelniej wyglada